File: backend/app/schemas/stat.py
from pydantic import BaseModel


class OverviewResponse(BaseModel):
    total_places: int
    active_places: int
    pending_places: int
    featured_places: int
    total_reviews: int
    pending_reviews: int
    total_users: int
    total_views: int
    average_rating: float


class TopPlaceResponse(BaseModel):
    id: int
    name: str
    ward: str
    total_views: int
    total_reviews: int
    average_rating: float
    favorite_count: int

File: admin-web/src/types/stat.ts
export interface OverviewResponse {
    total_places: number; active_places: number; pending_places: number;
    featured_places: number; total_reviews: number; pending_reviews: number;
    total_users: number; total_views: number; average_rating: number;
}
export interface TopPlaceResponse {
    id: number; name: string; ward: string; total_views: number;
    total_reviews: number; average_rating: number; favorite_count: number;
}
export interface KeywordStatResponse { keyword: string; count: number; }
export type TopPlaceOrderBy = 'VIEWS' | 'RATING' | 'REVIEWS' | 'FAVORITES';